src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -301,6 +328,16 @@ private void ProcessObjectContent(PSObject inputObject)
 
             if (inputBytes != null)
             {
+                Int32 targetLength = inputBytes.Length - (int)Offset;
```

Comment:
If `Offset` will be greater then `int.MaxValue`, then `(int)Offset` will result in negative value (or exception in debug configuration) and `targetLength` will be greater then `inputBytes.Length`. If `Offset` will be greater then `inputBytes.Length`, then `targetLength` will be negative as result.